client/src/components/Searchbar/Searchbar.js

```diff
@@ -3,21 +3,34 @@ import searchButton from "../../images/searchButton.png"
 import {useState} from 'react';
 import axios from 'axios';
 import { useProducts } from '../../features/Products/ProductsContext';
+import { useNavigate } from "react-router-dom";
+import { useSearchContext } from './searchContext';
+const baseURL = process.env.REACT_APP_API_URL;
 function Searchbar (){
     const [searchPhrase, setSearchPhrase] = useState('');
     const {setProducts} = useProducts();
+    const {setSearchRoute} = useSearchContext()
+    const navigate = useNavigate()
     const handleChange = (e) => {
         e.preventDefault();
         setSearchPhrase(e.target.value)
+        
 
     }
-    //not used in this deployment
+
+
     const handleClick = (e) => {
         e.preventDefault();
-        axios.get(`http://localhost:4000/search?q=${searchPhrase}`)
+        axios.get(`${baseURL}/search?query=${searchPhrase}`)
```

Comment:
what happens if you search for something like `fish&chips`? I think it will generate URL like this:
```
/search?query=fish&chips
```
So it will only search for `fish` because `&` is a query parameter separator. 

The `searchPhrase` should be URL encoded to prevent that happening.